File: src/main/java/org/openapitools/server/model/OfferConditionDTO.java
package org.openapitools.server.model;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import org.openapitools.server.model.OfferConditionQualityType;
import org.openapitools.server.model.OfferConditionType;
import jakarta.validation.constraints.*;
import jakarta.validation.Valid;


/**
 * Состояние уцененного товара. 
 */
public class OfferConditionDTO   {

    private OfferConditionType type;
    private OfferConditionQualityType quality;
    private String reason;

    /**
     * Default constructor.
     */
    public OfferConditionDTO() {
    // JSON-B / Jackson
    }

    /**
     * Create OfferConditionDTO.
     *
     * @param type type
     * @param quality quality
     * @param reason Описание товара. Подробно опишите дефекты, насколько они заметны и где их искать. 
     */
    public OfferConditionDTO(
        OfferConditionType type, 
        OfferConditionQualityType quality, 
        String reason
    ) {
        this.type = type;
        this.quality = quality;
        this.reason = reason;
    }



    /**
     * Get type
     * @return type
     */
    public OfferConditionType getType() {
        return type;
    }

    public void setType(OfferConditionType type) {
        this.type = type;
    }

    /**
     * Get quality
     * @return quality
     */
    public OfferConditionQualityType getQuality() {
        return quality;
    }

    public void setQuality(OfferConditionQualityType quality) {
        this.quality = quality;
    }

    /**
     * Описание товара. Подробно опишите дефекты, насколько они заметны и где их искать. 
     * @return reason
     */
    public String getReason() {
        return reason;
    }

    public void setReason(String reason) {
        this.reason = reason;
    }

    /**
      * Create a string representation of this pojo.
    **/
    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("class OfferConditionDTO {\n");
        
        sb.append("    type: ").append(toIndentedString(type)).append("\n");
        sb.append("    quality: ").append(toIndentedString(quality)).append("\n");
        sb.append("    reason: ").append(toIndentedString(reason)).append("\n");
        sb.append("}");
        return sb.toString();
    }

    /**
     * Convert the given object to string with each line indented by 4 spaces
     * (except the first line).
    */
    private static String toIndentedString(Object o) {
        if (o == null) {
          return "null";
        }
        return o.toString().replace("\n", "\n    ");
    }
}

File: lib/Models/OfferConditionDTO.cpp


#include "OfferConditionDTO.h"

using namespace Tiny;

OfferConditionDTO::OfferConditionDTO()
{
	type = OfferConditionType();
	quality = OfferConditionQualityType();
	reason = std::string();
}

OfferConditionDTO::OfferConditionDTO(std::string jsonString)
{
	this->fromJson(jsonString);
}

OfferConditionDTO::~OfferConditionDTO()
{

}

void
OfferConditionDTO::fromJson(std::string jsonObj)
{
    bourne::json object = bourne::json::parse(jsonObj);

    const char *typeKey = "type";

    if(object.has_key(typeKey))
    {
        bourne::json value = object[typeKey];




        OfferConditionType* obj = &type;
		obj->fromJson(value.dump());

    }

    const char *qualityKey = "quality";

    if(object.has_key(qualityKey))
    {
        bourne::json value = object[qualityKey];




        OfferConditionQualityType* obj = &quality;
		obj->fromJson(value.dump());

    }

    const char *reasonKey = "reason";

    if(object.has_key(reasonKey))
    {
        bourne::json value = object[reasonKey];



        jsonToValue(&reason, value, "std::string");


    }


}

bourne::json
OfferConditionDTO::toJson()
{
    bourne::json object = bourne::json::object();






	object["type"] = getType().toJson();






	object["quality"] = getQuality().toJson();





    object["reason"] = getReason();



    return object;

}

OfferConditionType
OfferConditionDTO::getType()
{
	return type;
}

void
OfferConditionDTO::setType(OfferConditionType  type)
{
	this->type = type;
}

OfferConditionQualityType
OfferConditionDTO::getQuality()
{
	return quality;
}

void
OfferConditionDTO::setQuality(OfferConditionQualityType  quality)
{
	this->quality = quality;
}

std::string
OfferConditionDTO::getReason()
{
	return reason;
}

void
OfferConditionDTO::setReason(std::string  reason)
{
	this->reason = reason;
}



